test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1

```diff
@@ -0,0 +1,345 @@
+        }
+
+        $numThreadJobs = (Get-Job | where PSJobTypeName -eq "ThreadJob").Count
+        $numThreadJobs | Should be 0
+    }
+
+    It 'ThreadJob Runspaces should be cleaned up at completion' {
+
+        try
+        {
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+            $rsStartCount = (Get-Runspace).Count
+
+            # Start four thread jobs with ThrottleLimit set to two
+            $Job1 = Start-ThreadJob -ScriptBlock { "Hello 1!" } -ThrottleLimit 5
+            $job2 = Start-ThreadJob -ScriptBlock { "Hello 2!" }
+            $job3 = Start-ThreadJob -ScriptBlock { "Hello 3!" }
+            $job4 = Start-ThreadJob -ScriptBlock { "Hello 4!" }
+
+            $job1,$job2,$job3,$job4 | Wait-Job
+
+            # Allow for clean to happen
+            Start-Sleep -Seconds 1
+
+            (Get-Runspace).Count | Should Be $rsStartCount
+        }
+        finally
+        {
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+        }
+    }
+
+    It 'ThreadJob Runspaces should be cleaned up after job removal' {
+
+        try {
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+            $rsStartCount = (Get-Runspace).Count
+
+            # Start four thread jobs with ThrottleLimit set to two
+            $Job1 = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 } -ThrottleLimit 2
+            $job2 = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 }
+            $job3 = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 }
+            $job4 = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 }
+
+            # Allow jobs to start
+            Start-Sleep -Seconds 1
+
+            (Get-Runspace).Count | Should Be ($rsStartCount + 4)
+
+            # Stop two jobs
+            $job1 | Remove-Job -Force
+            $job3 | Remove-Job -Force
+
+            (Get-Runspace).Count | Should Be ($rsStartCount + 2)
+
+        }
+        finally
+        {
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+        }
+
+        (Get-Runspace).Count | Should Be $rsStartCount
+    }
+
+    It 'ThreadJob jobs should work with Receive-Job -AutoRemoveJob' {
+
+        Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+
+        $job1 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } } -ThrottleLimit 2
+        $job2 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+        $job3 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+        $job4 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+
+        $null = $job1,$job2,$job3,$job4 | Receive-Job -Wait -AutoRemoveJob
+
+        (Get-Job | where PSJobTypeName -eq "ThreadJob").Count | Should Be 0
+    }
+}
+
+Describe 'Job2 Tests' -Tags 'CI' {
```

Comment:
I would rename the Describe to "Job2 class API tests" or something like this.
For an unfamiliar person it is not clear what 'Job2' is.

Reply:
Ok.